db/sensor.go

```diff
@@ -111,3 +119,45 @@ func (sm *sensorManager) ListByDevice(dvc string) (map[string]Sensor, error) {
 	}
 	return ret, err
 }
+
+func (sm *sensorManager) AddValue(id string, value *Value) error {
+	var key, err = getFullKey(id, sm)
+	if err != nil {
+		return err
+	}
+	json, err := json.Marshal(value)
+	if err != nil {
+		return err
+	}
+
+	return sm.db.cli.LPush("values:"+key, json).Err()
+}
+
+func (sm *sensorManager) RemoveValue(id string, start int, end int) error {
+	var key, err = getFullKey(id, sm)
+	if err != nil {
+		return err
+	}
+	return sm.db.cli.LTrim(key, int64(start), int64(end)).Err()
+}
+
+func (sm *sensorManager) GetValues(id string, start int, end int) ([]Value, error) {
+	var key, err = getFullKey(id, sm)
```

Comment:
It looks like `getFullKey()` should be a method of `sensorManager` - `func (sm *sensorManager) getFullKey(id string) (string, error)`